bitwarden_license/test/Commercial.Core.Test/SecretsManager/Queries/ServiceAccounts/CountNewServiceAccountSlotsRequiredQueryTests.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Bit.Commercial.Core.SecretsManager.Queries.ServiceAccounts;
+using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretsManager.Repositories;
+using Bit.Test.Common.AutoFixture;
+using Bit.Test.Common.AutoFixture.Attributes;
+using NSubstitute;
+using Xunit;
+
+namespace Bit.Commercial.Core.Test.SecretsManager.Queries.ServiceAccounts;
+
+[SutProviderCustomize]
+public class CountNewServiceAccountSlotsRequiredQueryTests
+{
+    [Theory]
+    [BitAutoData(5, 2)]
+    [BitAutoData(5, 0)]
+    [BitAutoData(5, 6)]
+    public async Task CountNewServiceAccountSlotsRequiredAsync_ReturnsCorrectCount(
+        int organizationSmServiceAccounts,
+        int currentServiceAccounts,
+        int serviceAccountsToAdd,
+        Organization organization,
+        SutProvider<CountNewServiceAccountSlotsRequiredQuery> sutProvider)
+    {
+        var availableServiceAccountSlots = Math.Max(0, organizationSmServiceAccounts - currentServiceAccounts);
+        var expectedAvailableServiceAccounts = Math.Max(0, serviceAccountsToAdd - availableServiceAccountSlots);
+
+        organization.SmServiceAccounts = organizationSmServiceAccounts;
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organization.Id)
+            .Returns(organization);
+
+        sutProvider.GetDependency<IServiceAccountRepository>()
+            .GetServiceAccountCountByOrganizationIdAsync(organization.Id)
+            .Returns(currentServiceAccounts);
+
+        var result = await sutProvider.Sut.CountNewServiceAccountSlotsRequiredAsync(organization.Id, serviceAccountsToAdd);
+
+        Assert.Equal(expectedAvailableServiceAccounts, result);
```

Comment:
The sut is called `CountNewServiceAccountSlotsRequired`, but the expected result is called `expectedAvailableServiceAccounts`. These are two different things though. Is this query telling us the number you need, or the number available?

Reply:
Its the number you need, I've revised it.